internal/handler/handler.go

```diff
@@ -25,14 +26,34 @@ func (handler *Handler) GetShortLink(ctx *gin.Context) {
 	originURL := string(body)
 
 	shortenURL := generator.GenerateRandomString(10)
-	if err := handler.store.SetLink(shortenURL, originURL); err != nil {
-		ctx.AbortWithStatus(http.StatusInternalServerError)
-		return
+
+	links := &store.StoreArgs{
+		ShortLink:    shortenURL,
+		OriginalLink: originURL,
+	}
+
+	res, err := handler.store.SetLink(links)
+
+	dbErr := &store.DuplicateValueError{}
+
+	if err != nil {
+		if !errors.As(err, &dbErr) {
```

Comment:
Я бы советовал использовать однострочную проверку вместо сложенных if.
```suggestion
	if err != nil && !errors.As(err, &dbErr) {
```